github-connector/cmd/github-connector/main.go

```diff
@@ -16,7 +16,9 @@ import (
 func main() {
 	log.Info("server started")
 
-	id, err := registration.RegisterService()
+	builder := registration.NewServiceDetailsBuilder(registration.NewOSCommunicator())
```

Comment:
You can consider the following solutions as an alternative to GITHUB_CONNECTOR_NAME env variable:
- name hardcoded
- name passed in arguments to the application

As a minimum I propose to read the env variable here and pass its value to NewServiceRegister method.